### PR TITLE
fix kind and ocp deploy and run on targets

### PR DESCRIPTION
**Pull request by msherif1234**
- fix deploy and run-on-kind targets
- allow to run make commands on MacOS BSD env
- add bundle check PR workflow job

Example:
```sh
BPFMAN_AGENT_IMG=quay.io/mmahmoud/bpfman-agent:pr-1150  BPFMAN_OPERATOR_IMG=quay.io/mmahmoud/bpfman-operator:pr-1150 make build-images

BPFMAN_AGENT_IMG=quay.io/mmahmoud/bpfman-agent:pr-1150  BPFMAN_OPERATOR_IMG=quay.io/mmahmoud/bpfman-operator:pr-1150 make run-on-kind 

oc get pods -n bpfman
NAME                               READY   STATUS    RESTARTS   AGE
bpfman-daemon-hdzqg                3/3     Running   0          9m29s
bpfman-operator-55d6cb6c57-fzw4f   2/2     Running   0          9m32s
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,4 @@ go.work
 
 # Temporary Kustomize file
 config/bpfman-deployment/kustomization.yaml
+config/bpfman-operator-deployment/kustomization.yaml
```
Comment (Billy99) [inline .gitignore:31]: No, you don't want this. This file is showing up as changed as a result of the `$(KUSTOMIZE) edit set image`, but that file is still under source control. I was thinking about replacing the `$(KUSTOMIZE) edit set image` with the `sed` command as well for that reason, but didn't because it is working as designed. The other file was just broken.

File: bundle/manifests/bpfman-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -215,7 +215,7 @@ metadata:
       ]
     capabilities: Basic Install
     containerImage: quay.io/bpfman/bpfman-operator:v0.0.0
-    createdAt: "2024-06-04T18:15:26Z"
+    createdAt: "2024-06-05T21:00:08Z"
```
Comment (Billy99) [inline bundle/manifests/bpfman-operator.clusterserviceversion.yaml:218]: I would just drop this change, meaningless.

File: bundle/manifests/bpfman-config_v1_configmap.yaml [hunk 3]
```diff
@@ -12,4 +12,6 @@ data:
     millisec_delay = 10000
 kind: ConfigMap
 metadata:
+  annotations: {}
+  labels: {}
```
Comment (Billy99) [inline bundle/manifests/bpfman-config_v1_configmap.yaml:16]: Why were these added?

**Review by Billy99 — COMMENTED** [review]
So does this build on the MAC? If so that is great. I never tried.

**Comment by msherif1234** [comment]
> So does this build on the MAC? If so that is great. I never tried.

this solves `sed` issue I don't have mac to see know for sure if that is it or there more issues, but if anyone has mac just install `gsed` and we should be good to go

File: bundle/manifests/bpfman-config_v1_configmap.yaml (hunk 3 above)
Comment (msherif1234) [reply bundle/manifests/bpfman-config_v1_configmap.yaml:16]: this change as well as TS change was side effect of `make bundle` will revert

File: .gitignore (hunk 1 above)
Comment (msherif1234) [reply .gitignore:31]: ok will remove it

File: bundle/manifests/bpfman-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (msherif1234) [reply bundle/manifests/bpfman-operator.clusterserviceversion.yaml]: removed

Comment (msherif1234) [reply bundle/manifests/bpfman-operator.clusterserviceversion.yaml]: dropped

**Review by Billy99 — APPROVED** [review]
/LGTM

Thanks
